apps/src/lab2/views/components/Instructions/NavigationArea.tsx

```diff
@@ -112,28 +115,99 @@ const NavigationArea: React.FC<NavigationAreaProps> = ({
     }
   }, [validationMessage, isRunning]);
 
-  const canShowButton = useMemo(() => {
+  // Certain labs hide the continue button until the user has met the conditions for the next level. Example: Music lab.
+  const isLabHidesContinueButton = useMemo(() => {
+    return LABS_HIDE_CONTINUE_BUTTON_RESOURCE_PANEL.includes(appName);
+  }, [appName]);
+
+  // For labs that hide the continue button, we show the button if the user has met the conditions for the next level.
+  // For labs that always show the continue button, this boolean is always true.
+  const showContinueButton = useMemo(() => {
```

Comment:
Along the lines of Molly's comment, the logic for disabling the continue button (for labs where it's always shown) and hiding it (for labs where it can be hidden) should be the same right? Could we just compute a single `isEnabled` boolean here, and then choose if that means to hide or disable it based on another prop?

Reply:
Oh, I just meant that regardless the lab, the computation for `hidden` and `disabled` should be identical, right? In other words, treating music lab as a special case, we just need to compute the conditions for `disabled` (regardless of lab, factoring in requireRun/hasRun, etc), and then specifically for Music Lab, that `disabled` becomes `hidden`?

Reply:
Gotcha - thanks. updated!